js/tenants.js

```diff
@@ -34,7 +34,31 @@ function getActiveTenant(cb) {
                 if (tenants.activeTenant) {
                     storage = chromeOrBrowser().storage.local;
                 }
-                return cb(tenants.activeTenant);
+
+                // first time the extension is invoked
+                // this will never have been set.
+                if (tenants.activeTenant === null) {
+                  return cb(null);
+                }
+
+                // if the activeTenant has a region property
+                // then the active tenant is an object
+                if (tenants.activeTenant && tenants.activeTenant.region) {
+                  return cb(tenants.activeTenant);
+                } else {
+                  // if the user had previously saved the institution
+                  // the activeTenant is a short code string
+                  // we retreive the up to date tenant list from s3
+                  // and replace the string with a tenant object
+                  getTenants(function(updatedTenantsList) {
```

Comment:
In all other cases we `return cb(...)` but this here won't return a value - what does that callback return and will it be a problem that this line isn't returning?

Reply:
agreed, im returning null there which will force the user to the options pane

Reply:
I meant that the return you have here is inside a callback, which means current function (`function(tenants) {`) won't return a value in this branch of code.